arraykit.c

```diff
@@ -230,6 +287,23 @@ row_1d_filter(PyObject *Py_UNUSED(m), PyObject *a)
     return a;
 }
 
+//------------------------------------------------------------------------------
+// array utility
+
+// Specialized array deepcopy that stores immutable arrays in memo dict.
+static PyObject *
+array_deepcopy(PyObject *Py_UNUSED(m), PyObject *args)
+{
+    PyObject *array, *memo;
+    if (!PyArg_ParseTuple(args, "OO:array_deepcopy",
+            &array,
+            &memo))
+    {
```

Comment:
```suggestion
    if (!PyArg_UnpackTuple(args, "array_deepcopy", 2, 2, &array, &memo)) {
```

Reply:
I see now how `PyArg_UnpackTuple` is better here; thanks!